### ISSUE TITLE
Sometimes takeoff fails in CI tests

### ISSUE THREAD

**Issue by julianoes** [PX4/PX4-Autopilot#15450]
When working on the MAVSDK SITL CI tests I often run into the case where a takeoff fails completely, or the landed state toggles back and forth between finally taking off.

Here are examples where takeoff doesn't happen at all:
https://logs.px4.io/plot_app?log=0f20b944-0704-4cbf-b499-1aec5bf6f9d6
https://logs.px4.io/plot_app?log=690f6488-0644-4b5c-a9cb-8ccdcda7f3ae

![bokeh_plot](https://user-images.githubusercontent.com/1419688/88920913-edbbd680-d26d-11ea-91de-f67a65c7d563.png)

And here are examples where takeoff on the second try seems to work:
https://logs.px4.io/plot_app?log=972ffaf8-da85-4142-a469-7195e0ba342f
https://logs.px4.io/plot_app?log=f13ba39e-cab0-4feb-8f6a-f7b8a506bc9c
https://logs.px4.io/plot_app?log=bab77966-e501-4923-9181-e922063a4109

You can see how there is:
```
[commander] Takeoff detected
[commander] Landing detected
[commander] Takeoff detected
```

FYI @MaEtUgR @dagar

**Comment by MaEtUgR** [comment]
https://github.com/PX4/Firmware/issues/15408

**Comment by MaEtUgR** [comment]
This was solved by #15509
